### DISCUSSION TITLE
需求 & 项目结构探讨

### DISCUSSION THREAD

**Discussion by LordFoxFairy** [fp-space/MiniNote#4] [Ideas]
1. 需求
2. 项目结构

**Comment by LordFoxFairy** [comment]
@hexWars

**Reply by LordFoxFairy** [reply]
1. 页面布局设计
1.1 左边目录树
1.2 中间markdown笔记渲染 + 右边内容目录索引
1.3 上方导航栏设计

2. 加载存放 markdown 内容 & 文件夹

3. 搜索框，如何快速搜素定位到某一块内容

4. 仓库项目结构整理构思，为每一个模块定义，以及规定

**Reply by LordFoxFairy** [reply]
第4点，可以一起探讨，整理，直到最终成型

**Reply by hexWars** [reply]
目录树是什么概念，是不是类似[这里](https://docs.astro.build/en/getting-started/)的左边，例如，markdown存放目录是下面这样的，而左边的技术就是一个可折叠栏，点开下面是文章

```
├── 技术/
│   ├── nginx使用指南
│   └── docker使用指南
├── 生活/
│   ├── 周记1
│   └── 周记2
```

如果是这样的话，有几个问题，最多允许嵌套多少层？目录树文件夹一层，文章两层，这样效果为佳

---

还有一种简要的写法，就是不管层级，按照时间顺序排列，例如

```
├── 技术/
│   ├── nginx使用指南 // 2024-1-1
│   └── docker使用指南 // 2024-1-3
├── 生活/
│   ├── 周记1 // 2024-1-4
│   └── 周记2 // 2024-1-2
```

那么左边可以显示成

```
─ nginx使用指南 // 2024-1-1
─ 周记2 // 2024-1-2
─ docker使用指南 // 2024-1-3
─ 周记1 // 2024-1-4
```

**Reply by hexWars** [reply]
标签和分类，一般是写在`Front Matter`里面，也就是markdown前面的那个东西，下面是例子：
```
---
title: sp-space 文档
pubDate: 2025-01-01
tags: ["周记", "css","tailwindcss"]
categories: ["生活"]
---

最近XXX
```

tag没啥好说的，就是直接写

categories又有几种分法，是直接在`Front Matter`里写，还是根据文件实际所属的目录来分

或者干脆写简单版的，只弄tag，别的先不管

**Reply by hexWars** [reply]
关于markdown的渲染，用rust解析器解释成html后，可以直接使用tailwindcss的一个排版插件，这个应该很容易做到

**Reply by hexWars** [reply]
关于搜索框，这个其实不一定要弄，现在的很多博客/技术文档，用的是[这个](https://www.algolia.com/)，直接在markdown里写好怎么集成就行了

> 因为这个服务，针对技术博客和技术文档，免费使用

当然，用一些搜索集成也可以

**Reply by LordFoxFairy** [reply]
语法按照markdown来，不要修改别人记录，层级按照个人创建的层级来展示

**Comment by LordFoxFairy** [comment]
1. 不要修改别人源文件
2. 标签单独可设置一个可配置目录，或者个人习惯在笔记里面设置按照markdown格式
3. 层级不做限制

**Reply by hexWars** [reply]
你希望，纯markdown格式，不加任何的`Front Matter`对吧。这样的话，文章就无法显示创建时间了（如果按照md文件的创建和编辑时间来定的话，时间一定是dist目录build时候的时间）

tag的话，我没懂，单独设置可配置目录的意思

层级不限制的话，侧边可能不太美观，不过先这样干吧，按照真实目录的层级对吧

**Reply by LordFoxFairy** [reply]
1. 在markdown可做渲染先
2. 层级不限制，要符合个人笔记习惯，不能强迫别人

**Reply by hexWars** [reply]
好吧hhh，我只是在说业内主流的方案（很多文档主题都是这样干的），不过这个`Front Matter`确实没有一个统一的标准

**Reply by LordFoxFairy** [reply]
我先弄下文件树这个吧，左边的那一块，数据我弄假数据先 @hexWars 
需求：文件树

**Reply by LordFoxFairy** [reply]
@hexWars https://github.com/ranile/material-yew 我们统一采用这个框架吧，没专业前端写css太过于困难了，还是用基于 tailwind的 框架来比较适合

**Reply by hexWars** [reply]
组件库啊，可以，那我先把顶部导航栏弄一下吧，我的设想是，顶部导航栏的数据，通过一个config文件来获取，我先做单级导航栏
@LordFoxFairy

**Comment by LordFoxFairy** [comment]
wasm 的限制，我发现很多第三方库，用不了，我想了下，我把文件树的搜索给去掉，我感觉都静态文件部署了，这功能去掉，
后续，可以部署web，可以弄进去，等这个弄完，就弄一个beta版本realse吧，发布出来，我们就开始着手 桌面版如何 @hexWars

**Reply by LordFoxFairy** [reply]
你先看下现在的整体逻辑，我们先同步下思路 @hexWars

**Reply by hexWars** [reply]
晚点我再看看具体的逻辑，这个昨天还没看，至于桌面版，你是希望说桌面版点击后网页就同步发布还是怎么样

桌面版的UI具体用哪个？这块你写过嘛，我经验为0🤣

**Reply by LordFoxFairy** [reply]
桌面我也没写过，但是我看了下，基本是基于electron，但是基于 chrome 内核，过于笨重，我考虑了下，还是采取 tauri，这个已经支持 **rust**，而且我看了demo，感觉还不错，只是用的原生的 webview 有很多兼容性，但是对我们来说，不需要考虑到这 @hexWars

**Reply by LordFoxFairy** [reply]
@hexWars 获取当前点击的文件已经实现了，可以浏览器控制台上看到打印日志

**Reply by hexWars** [reply]
你是不是没上传

**Reply by LordFoxFairy** [reply]
![image](https://github.com/user-attachments/assets/d3eb83ad-36f1-4888-ad1a-dc901a0e3a69)

@hexWars 已经上传了，你需要看下 浏览器的控制台

**Reply by LordFoxFairy** [reply]
https://mathjax.github.io/MathJax-demos-web/speech-tex-chtml.html

@hexWars 我看是用的这个

**Reply by hexWars** [reply]
这两天不太有空，你要不先休息一下，我到现在力扣都还没写

**Comment by LordFoxFairy** [comment]
1. 目前，大纲未做
2. 布局样式
3. latex 语法矩阵渲染不正确，以及其他部分【可能语法写错】
4. 代码重构

**Reply by hexWars** [reply]
似乎是它没有准确读取到数学公式，例如
```
$$
A = [a_{ij}]_{N\times N}
$$
```
修改成
`$$A = [a_{ij}]_{N\times N}$$`
就可以识别了

或者
```
$$
​	$A$是状态转移概率矩阵：
$$
A = [a_{ij}]_{N\times N}
$$
其中，// 似乎是把这里识别成了数学公式，太密集了，上面和下面加个换行就行了
$$
a_{ij} = P(i_{t+1}=q_j|i_t=q_i)，i=1,2,\cdots,N;j=1,2,\cdots,N
$$
```

**Reply by hexWars** [reply]
我再改改样式吧，这里又溢出了

![image](https://github.com/user-attachments/assets/c7b13826-1e1c-4ca5-849e-b4c8508576d1)

**Comment by LordFoxFairy** [comment]
强，latex的展示，我看了下，有几个，
1. 这个很长的数学公式，本来我是换行的，但是这里渲染没给换行
2. latex 渲染矩阵数学公式 没有呈现矩阵形式，而是一行

latex剩下这两个，

我先弄下大纲这里下，很快的 @hexWars

**Reply by hexWars** [reply]
可否直接给个例子

**Reply by LordFoxFairy** [reply]
@hexWars 新上传几个例子，可以查看到了

但是 latex 现在没生效了，我等下回去看下

**Reply by LordFoxFairy** [reply]
latex 我找到原因了，我本地升级版本太新了，语法不一样了 ，绷不住了，还好我没提交，现在就差一些报错了，我发现，script.rs的方式太粗暴了，我每次都要重新渲染，你看下我们如何优化下

**Reply by LordFoxFairy** [reply]
@hexWars 我知道的，gitbook，有些不支持，
![Uploading image.png…]()


我的想要的是这样，这样的话，应该才是正确的渲染效果，

**Reply by LordFoxFairy** [reply]
![image](https://github.com/user-attachments/assets/3a4ca839-83db-4bee-88fe-ff689a3170ad)

**Reply by LordFoxFairy** [reply]
目前我将大纲弄上去了，但是样式惨不忍睹，还有大纲选中后锚定到对应标题，这个还没实现。 代码重构需要等待基本功能完成后，梳理下，代码结构肯定要重新弄下

**Reply by LordFoxFairy** [reply]
> ![image](https://private-user-images.githubusercontent.com/34004635/398535836-3a4ca839-83db-4bee-88fe-ff689a3170ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUxMTUzMjMsIm5iZiI6MTczNTExNTAyMywicGF0aCI6Ii8zNDAwNDYzNS8zOTg1MzU4MzYtM2E0Y2E4MzktODNkYi00YmVlLTg4ZmUtZmY2ODlhMzE3MGFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI1VDA4MjM0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyZWUyZWU0OTg3ODMxOGY0OTA4N2RjZDM2NmZiYzUwMmI5ZDBhNDE2YmMzYjc4ZDljYjI2MmU5M2UxNmY1NjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZKsItv4PkRzg3vCxAJqAnpwBfHAnO5WwTJcpzcxb7wo)

在这里 @hexWars

**Reply by hexWars** [reply]
那个换行的我搞定了，是转义问题，无语了，搞半天latex的配置

**Reply by hexWars** [reply]
不过可能有更好的解决方案就是了

**Reply by LordFoxFairy** [reply]
@hexWars 强，现在还差点就可以弄完了，
1. 大纲
2. 代码重构
3. 样式整体调整，抽出样式，放到scss
